pandas/_libs/tslibs/offsets.pyx

```diff
@@ -2476,12 +2476,38 @@ cdef class MonthOffset(SingleConstructorOffset):
 cdef class MonthEnd(MonthOffset):
     """
     DateOffset of one month end.
-
+    
+    When parameter n=0, always offset to the end of month.
+    
+    When parameter n=1, depend on the given date:
+        a) Given date is on an anchor point (last day of the month) -> offset to the end of the following month.
+        b) Given date is not on an anchor point -> offset to the end of the same month.
+        
+    Parameters
+    ----------
+    n : int, default 1
+    
     Examples
     --------
     >>> ts = pd.Timestamp(2022, 1, 1)
-    >>> ts + pd.offsets.MonthEnd()
+    >>> ts + pd.offsets.MonthEnd(n=1)
     Timestamp('2022-01-31 00:00:00')
+    
```

Comment:
can you add a short sentence before the examples you've added to clarify what the reader is meant to take away from them? see https://pandas.pydata.org/docs/development/contributing_docstring.html#section-7-examples